File: backend/src/models/equipments.py
from sqlalchemy import ForeignKey, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from src.models.mixins import IntPkBase as Base


class Equipment(Base):
    __tablename__ = "equipments"

    name: Mapped[str] = mapped_column(String(255), unique=True)
    image: Mapped[str] = mapped_column()

    type: Mapped["EquipmentType"] = relationship(back_populates="equipments")

    type_id: Mapped[int] = mapped_column(ForeignKey("equipment_types.id"))


class EquipmentType(Base):
    __tablename__ = "equipment_types"

    name: Mapped[str] = mapped_column(String(255), unique=True)

    equipments: Mapped[list[Equipment]] = relationship(back_populates="type")

File: frontend/src/types/equipments.ts
export interface CreateEquipment {
  name: string;
  image: string;
  type_id: number;
}

export interface SimpleEquipment extends CreateEquipment {
  id: number;
}

export interface Equipment {
  id: number;
  name: string;
  image: string;
  type: SimpleEquipmentType;
}

export interface CreateEquipmentType {
  name: string;
}

export interface SimpleEquipmentType extends CreateEquipmentType {
  id: number;
}

export interface EquipmentItem {
  id: number;
  name: string;
  image: string;
}

export interface EquipmentType extends SimpleEquipmentType {
  equipments: EquipmentItem[];
}
